packages/plugin-cosmos/src/actions/cosmosTransfer.ts

```diff
@@ -0,0 +1,204 @@
+import { elizaLogger } from "@ai16z/eliza";
+import { transferTemplate } from "../templates";
+import { DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+import { SigningCosmWasmClient } from "@cosmjs/cosmwasm-stargate";
+import { z } from "zod";
+import { Transaction } from "../types";
+import { PaidFee } from "../services/paid-fee";
+import { AssetsPicker } from "../services/assets-picker";
+import { assets } from "chain-registry";
+import { AssetsAdapter } from "../services/assets-adapter";
+import { FeeEstimator } from "../services/fee-estimator";
+
+export { transferTemplate };
+
+const cosmosTransferParamsSchema = z.object({
+    denomOrIbc: z.string(),
+    amount: z.string(),
+    toAddress: z.string(),
+});
+
+export type CosmosTransferParams = z.infer<typeof cosmosTransferParamsSchema>;
+
+export class TransferActionParamsValidator {
+    validate(params: any): CosmosTransferParams {
+        try {
+            const validParams = cosmosTransferParamsSchema.parse(params);
+
+            return validParams;
+        } catch (error) {
+            elizaLogger.error(JSON.stringify(error, undefined, 4));
+        }
+    }
+}
+
+export class TransferAction {
+    private walletProvider: DirectSecp256k1HdWallet;
+    readonly rpcEndpoint: string;
+    readonly chainName: string;
+
+    constructor(
+        walletProvider: DirectSecp256k1HdWallet,
+        rpcEndpoint: string,
+        chainName: string
+    ) {
+        this.walletProvider = walletProvider;
+        this.chainName = chainName;
+        this.rpcEndpoint = rpcEndpoint;
+    }
+
+    async transfer(params: CosmosTransferParams): Promise<Transaction> {
+        console.log(
+            `Transferring: ${params.amount} tokens to ${params.toAddress}`
+        );
```

Comment:
?

Reply:
I can remove it, this is just how it was done in `evm-plugin`.  https://github.com/blockydevs/eliza/blob/2263d767721d463b2575892fb6c2ec879c800a39/packages/plugin-evm/src/actions/transfer.ts#L23